src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+  }
+
+  @Test
+  /** Verifies that the completed task shows as completed on the helper's user profile */
+  public void _08_Userpage_AsHelper_CompletedTask() {
+    logOut("logout-href");
+    ifLaggingThenRefresh();
+    loginUser(USER_EMAIL_HELPER);
+    ifLaggingThenRefresh();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+
+    // Helper's total points
+    String points =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("points"));
+                  }
+                })
+            .getText();
+    assertEquals("My current points: " + Integer.toString(helperPoints) + "pts", points);
+    goToOfferHelp();
+    String completedTaskStatus =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(
+                        By.xpath("//tbody[@id='complete-task-body']/tr[1]/td[2]"));
+                  }
+                })
+            .getText();
+    // Verifies that the task got updated to the COMPLETED status
+    assertEquals(recentTask.get("status"), completedTaskStatus);
+  }
+
+  @Test
+  /** Test functionality of disapproving a helper's completed task */
+  public void _09_Userpage_AsLoggedUser_DisapproveTask() {
+    backToHome();
+    ifLaggingThenRefresh();
+    updateRecentTask(); // updates the recent task we have saved with the most recent task in the
+    // homepage
+    helpOut();
+    goToUserPage();
+    ifLaggingThenRefresh();
+    goToOfferHelp();
+    completeTaskAsHelper();
+    logOut("logout-href");
+    ifLaggingThenRefresh();
+    loginUser(USER_EMAIL);
+    ifLaggingThenRefresh();
+
+    // click on disapprove task button.
+    // this is located in row 2 for this order of events but we probably need a better sorting
+    // system for completed tasks so that it sorts by time of completion (making it easier to
+    // find where the most recently approved/completed task is and easier to test in the future)
+    wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(
+                    By.xpath("//tbody[@id='await-verif-body']/tr[2]/td[5]/button"));
+              }
+            })
+        .click();
+    // Clicking disapprove triggers an alert confirmation window
+    // Driver will try to accept the alert to verify the task every second for a minute
+    for (int i = 0; i < 60; i++) {
+      try {
+        driver.switchTo().alert().accept();
+        break;
+      } catch (NoAlertPresentException e) {
+        driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS);
+      }
+    }
+    // updates recentTask
+    recentTask.put("status", "IN PROGRESS");
+
+    String taskStatus =
```

Comment:
Same suggestion here: Using a loop to replace these three blocks of code can be a good idea